### PR TITLE
additional-properties-add-test-cases-for-spread-of-record

### PR DESCRIPTION
**Pull request by qiaozha**
fix https://github.com/Azure/cadl-ranch/issues/546

# Cadl Ranch Contribution Checklist:

- [ ] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [ ] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [ ] I have written a [mock API](../docs/writing-mock-apis.md)
- [ ] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: packages/cadl-ranch-specs/http/type/property/additional-properties/main.tsp
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/type/property/additional-properties/main.tsp:323]: Why we use `knownProp: ModelForRecord[];` here? I assume we could use any type (not limited to `ModelForRecord[]`)?

Comment (qiaozha) [reply packages/cadl-ranch-specs/http/type/property/additional-properties/main.tsp:323]: we are, but I was trying to show that if the known properties type is the same as the additional properties type, we can have three ways to express that. 1, extends Record, 2. is Record, 3. spread of Record. 
I have another case here to show different types here https://github.com/Azure/cadl-ranch/pull/556/files#diff-9ab37e973a0f0d7e61ea487e765530d01f107dc0e5fec04b9dcbc829acd8a48aR375-R387

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/type/property/additional-properties/main.tsp:323]: Sure. Just wondering if we only want to show that "known properties type is the same as the additional properties type", we may use simpler type (instead of an array).

No strong opinion here. Just I don't know if array works for current Java emitter :-(

**Review by weidongxu-microsoft — APPROVED** [review]
I did a quick look and nothing seems wrong.

Count me as half of one approver.

@haolingdong-msft @XiaofeiCao from Java as well, if see anything apparently not good, please point out.

**Comment by msyyc** [comment]
Python has no work for additional properties so it is better to get approval from JAVA/.NET

**Comment by haolingdong-msft** [comment]
wonder do we have service using this spread of `Record`?

File: packages/cadl-ranch-specs/http/type/property/additional-properties/main.tsp
Comment (haolingdong-msft) [inline packages/cadl-ranch-specs/http/type/property/additional-properties/main.tsp:261]: why do we add the property?

Comment (qiaozha) [reply packages/cadl-ranch-specs/http/type/property/additional-properties/main.tsp:261]: because previous extends and is Record way only works when the known property type is the same as the additional properties type.

**Review by haolingdong-msft — APPROVED** [review]

**Pull request merged**
